Framework/GenAIFramework/Code/Include/GenAIFramework/GenAIFrameworkTypes.h

```diff
@@ -30,4 +31,8 @@ namespace GenAIFramework
     using ModelAPIResponse = AZ::Outcome<AZStd::string, AZStd::string>; //!< The type of response from the model provider
     using ModelAPIExtractedResponse = AZ::Outcome<AIMessage, AZStd::string>; //!< The type of extracted response from the model
     using AIHistory = AIMessages; //!< The type of AI history
+
+    using FeatureTuple = AZStd::tuple<AZStd::string, AZStd::string, AZStd::string>; //!< The tuple of model, provider, and feature names
+    using FeaturesConversationsStore = AZStd::unordered_map<AZStd::string, FeatureTuple>; //!< The store of feature conversations
```

Comment:
I would rename it to `FeaturesConversationsMap` to keep it consistent with the other names